module/zfs/mmp.c

```diff
@@ -322,9 +323,12 @@ mmp_write_uberblock(spa_t *spa)
 	int label;
 	uint64_t offset;
 
+	spa_config_enter(spa, SCL_STATE, FTAG, RW_READER);
 	vd = vdev_random_leaf(spa);
-	if (vd == NULL || !vdev_writeable(vd))
+	if (vd == NULL || !vdev_writeable(vd)) {
```

Comment:
Good catch.  Let's batch up some of this very minor tweaking without any functional changes for a followup PR.